File: bookmark/bkmapp-fr/js/boundary/userstore.js
import { postJsonData } from "../service/restcall.js";
import configData from "../service/config.js";

const url = `${configData.baseurl}/users`

const doLogin = async (usr, pwd) => {
    return postJsonData(`${url}/login`,{usr,pwd},false);
}

const doRegistration = async (usr, pwd, fname, lname, ruole) => {
    const UserData = {
        first_name:fname,
        last_name:lname,
        email:usr,
        pwd:pwd,
        roleuser:ruole
    };
    return postJsonData(url,UserData,false);
}


export {doLogin, doRegistration};

File: Flutter/untitled/lib/control/UserStore.dart
import '../entity/user.dart';
import '../service/rest.dart';

class userStore {

  static Future getUser(String usr, String pwd) async {
    dynamic data = {'usr': usr, 'pwd': pwd};
    late User _user;
    try {
      var resp = await rest.postRest("/users/login", false, "", data);
      _user = User.fromJson(resp);
      return (_user);

    } catch (e) {
      return User(
          error: e.toString(),
          firstName: "",
          lastName: "",
          mail: "",
          role: "",
          token: "",
          userid: 0);
    }
  }

  static Future posUser(
      String firstName, String lastName, String mail, String pwd) async {
    dynamic data = {
      'first_name': firstName,
      'last_name': lastName,
      'email': mail,
      'pwd': pwd,
      'roleuser': 'User'
    };
    late User _user;
    try {
      var resp = await rest.postRest("/users", false, "", data);
      _user = User.fromJson(resp);
      return (_user);


    } catch (e) {
      return User(
          error: e.toString(),
          firstName: "",
          lastName: "",
          mail: "",
          role: "",
          token: "",
          userid: 0);
    }
  }
}
